### PR TITLE
PR: Add workflow that tests different spyder versions

### PR DESCRIPTION
**Pull request by skjerns**
Coming from the previous PR #56 I figured it might make sense to have CI workflows that test for different versions of spyder.

Conda gives some tight constraints on which packages to install, so I opted to simply install the spyder versions with pip in this workflow. Let's see if this works. Obviously they will not work with Spyder 5.X until #56 has been merged.

### CODE REVIEW

**Comment by skjerns** [comment]
perfect, seems to work :) now just need to merge #56 and then we can check compatibility with different Spyder versions.

**Comment by dalthviz** [comment]
This seems like an interesting idea @skjerns ! Maybe we could somehow even run some integration tests where we launch Spyder and check things like for example the plugin loading or simulate user interaction with the interface via pytest-qt/qtbot to run the profiler inside Spyder.

Just in case what do you think @jitseniesen @ccordoba12 ?

**Comment by ccordoba12** [comment]
I don't know if it's worth the effort to test on older Spyder versions because we don't maintain them anymore. But if you think this is important, please go for it.

**Comment by skjerns** [comment]
Now that #56 is merged, do you think we could integrate this additional testing? @dalthviz

**Comment by dalthviz** [comment]
We could add it I think (having some integration testing sounds nice to me). However, I think we should then check a way to run Spyder and check the interaction with the plugin with the Spyder instance running, something like the [`test_mainwindow.py`](https://github.com/spyder-ide/spyder/blob/5.x/spyder/app/tests/test_mainwindow.py#L2436) that Spyder has but with tests to check `spyder-line-profiler` (checking for example if the run action is in the Run menu, loading and profiling and example file through the shortcut, etc).

Also, I think it could be nice to do the tests with the latest Spyder 5 patch/bug fix versions so `5.1.5`, `5.2.2` and `5.3.0` (`5.3.1` when it gets released).

What do you think @skjerns ?

File: .github/workflows/linux-tests-spyder-versions.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    name: Linux Py37 Spyder Version
+    runs-on: ubuntu-latest
+    env:
+      CI: True
+      SPYDER_VERSION: ${{ matrix.SPYDER_VERSION }}
+      RUNNER_OS: 'ubuntu'
+    strategy:
+      fail-fast: false 
+      matrix:
+        SPYDER_VERSION: ['5.1.5', '5.2.0', '5.3.0']
```
Comment (dalthviz) [inline .github/workflows/linux-tests-spyder-versions.yml:22]: Update Spyder versions to the latest patches:
```suggestion
        SPYDER_VERSION: ['5.1.5', '5.2.2', '5.3.2']
```

**Pull request closed without merging**